core/base/v7/inc/ROOT/TDrawable.hxx

```diff
@@ -34,7 +35,11 @@ class TDrawable {
 public:
   virtual ~TDrawable();
 
-  virtual void Paint(TVirtualCanvasPainter& onCanv) = 0;
+  virtual void Paint(TVirtualCanvasPainter &) = 0;
+
+  virtual void FillMenu(TVirtualCanvasPainter &) = 0;
+
+  virtual void ExecMenu(const std::string &) = 0;
```

Comment:
Name ExecuteMenuItem() gives an impression that TMenuItem will be provided as argument - but this is not a case. (vector of) TMenuItem is temporary object and destroyed immediately after info is delivered to the JavaScript client.
I will use name Execute() which means execute given method with given arguments as is coded in provided string.